suse-open-source-policy.html

```diff
@@ -179,13 +179,23 @@ <h3 id="managing-source-code">Managing Source Code</h3>
 
     <h3 id="factory-first">Factory First</h3>
 
-    <p>When packaging new Open Source software or adding changes to existing packages follow the “Factory First” model. “Factory” is the common code stream all our distributions are based on. It is the immediate source for openSUSE Tumbleweed and it eventually ends up in openSUSE Leap and the SUSE Linux Enterprise distributions. This is a continuation of the “Upstream First” principle on the distribution level. It reduces maintenance effort and leverages the community.</p>
+    <p>When packaging new Open Source software or adding changes to existing packages follow the "Factory First" model. <a href="https://susedefines.suse.com/definition/opensuse-factory/">"Factory"</a> is the common code stream all our distributions are based on. It is the immediate source for openSUSE Tumbleweed and it eventually ends up in openSUSE Leap and the SUSE Linux Enterprise distributions. This is a continuation of the "Upstream First" principle on the distribution level. It reduces maintenance effort and leverages the community.</p>
 
-    <p>“Factory First” extends to other SUSE products. Packages for all products should generally come from the common code stream our distributions are based on. So for packages you need in any products contribute them to factory first.</p>
+    <p>"Factory First" extends to other SUSE products. Packages for all products should generally come from the common code stream our distributions are based on. So for packages you need in any products contribute them to factory first.</p>
+
+    <h3 id="first-summary">"First" summary</h3>
+
+    <p>"Upstream first" applies to all individual upstream projects, such as the Linux kernel and nearly any package on SLE.</p>
+
+    <p>"Upstream first" also applies when there is an estalished upstream distribution, such as Tumbleweed (for SUSE Linux Enterprise) or Uyuni (for SUSE Manager)</p>
+
+    <p>"Factory first" applies to the SUSE Linux Enterprise family. It is recommended for things tightly connected to SUSE Linux Enterprise. It should also be considered (not necessarily adopted) for every SUSE product.</p>
```

Comment:
It is not only recommended for things tightly connected to SUSE Linux Enterprise, but in my opinion it should also clearly state that it is mandatory for submissions to the SLE code base. Without following Factory First on SLE submissions the submission will be rejected, with the exception of very few submissions (like for example Metadata packages)

Reply:
Agreed. We actively encourage openSUSE to span a broader scope than just SLE in order to give us benefits for all of our products. We shouldn't risk giving the impression we're abandoning our reliance on our community contributions.

Reply:
Why would you want to leave it open for any SUSE product to _not_ adopt this policy?

Reply:
Please note there is a typo in the first line: estalished should be established

Reply:
> Agreed. We actively encourage openSUSE to span a broader scope than just SLE in order to give us benefits for all of our products. We shouldn't risk giving the impression we're abandoning our reliance on our community contributions.

Encouraging openSUSE to expand its horizons is a noble goal, although I don't see it happening. Forcing the openSUSE development model on every project that SUSE embarks on is plain wrong. So while the SUSE Open Source policy is applied to the whole company, it was never the intent to have all of SUSE's projects integrated in SLE and/or openSUSE and by extension Factory.

Reply:
> Encouraging openSUSE to expand its horizons is a noble goal, although I don't see it happening

You don't see the fact that Tumbleweed now has almost 14000 source packages, and variants such as MicroOS and Kubic, as evidence that openSUSE is not expanding it's horizons?

I'm sorry @vojtech-pavlik but I have to defend openSUSE here..it's been encouraged to expand it's horizons, and it has.

I'm sure it could do more, but we shouldn't dismiss the successes it has just because it is uncomfortable evidence against the desired narrative.

Reply:
@vojtech-pavlik As one of those folks who has been heavily driving improvements across the board in openSUSE, I think I feel somewhat slighted here.

I've been doing a ton of work in openSUSE as a distribution, with some prominent examples:

* Improving how SUSE rpm is packaged (I created `rpm-config-SUSE`, which apparently just was backported to SLE 15 SP3)
* Fixing the user experience with transactional updates for MicroOS, and developing better integration for MicroOS Desktop
* Launching [openSUSE Code](https://code.opensuse.org) to explore Git-oriented workflows for the openSUSE distributions (especially packaging) and giving people a fully free software platform to do project management and code hosting
* Working on improving security and identity management offerings in openSUSE by working on SELinux support, working with @hellcp to bring FreeIPA to openSUSE, and packaging Ipsilon for openSUSE

And that's on top of being a member of the openSUSE board. Now, I could understand that this isn't all immediately visible to you. If there's a visibility problem, we should address that. But saying the project is failing to expand its horizons is definitely wrong.

Reply:
@Conan-Kudo Thank you for all the work and contributions, they're worthwhile for the project and I do very much appreciate them even personally, I know that eg. SELinux policies have been long neglected and making git more central to openSUSE (and SLE/LEO) development is also a direction we need to take.

But they're not the new horizons we talked about when discussing that all SUSE projects should go to Factory. That would be entirely new projects, not related to an operating system, coming under the openSUSE umbrella. So, @sysrich No, 14000 packages are not that, that's growth of the very same concept of packaging things into a distro. Kubic is perhaps the closest, as, while an OS again, it's one with a well defined usecase that's different of that of what Tumbleweed/Leap offer.

What I wanted to say, and again, no offense intended, is that SUSE will work on projects that aren't even fitting into "let's package that" scheme - and so pushing them through Factory will not make sense. They will be open source and will have their own upstreams that best suit their needs. And I won't be asking openSUSE to morph into an Apache Foundation to enable any project either - I think it's a great community a great project for what it is.

Reply:
> What I wanted to say, and again, no offense intended, is that SUSE will work on projects that aren't even fitting into "let's package that" scheme - and so pushing them through Factory will not make sense. They will be open source and will have their own upstreams that best suit their needs. 

@vojtech-pavlik My question is..why?

By my reckoning  (and www.opensuse.org confirms) openSUSE currently homes several projects that do not fit your definition of "let's package that"

They include

- Kubic (Kubernetes optimised distro)
- MicroOS (also a very well defined usecase that's different from Tumbleweed/Leap)
- openQA
- OBS

All of these projects have had significant SUSE contributions following a "Factory First" approach as much as is applicable for their comparable SUSE products/self-use of the Projects.

All of the above have been successful in their own right, and that success has been used by SUSE to sing its own praises about it's contributions and the benefits those Projects bring to the SUSE eco system.

This was codified in the versions of this policy lower than 1.0.3. Communities have built around that. Customers, Partners, and external stakeholders have taken it as read that SUSE would contribute in that manner (because literally, they could read it).

Now that has changed.

And explanation has not been given as to why.

Your posts to date have been very matter of fact about what SUSE sees it's route to be now, with less emphasis on openSUSE, Factory First, and the planks which it saw important enough to codify in this policies earlier versions.

I'm certainly not going to argue with a VP as to the merits (or lack there of) of a SUSE business decision.

But that business decision does have an impact on SUSE's communities and that business decision does manifest itself in this proposed version 1.0.3.

And I believe the stakeholders of that policy, namely the openSUSE community, deserve a well explained justification for that change. I do not think "SUSE have decided to do things differently" is a good one, especially when there is a clear body of work, in the public, that was working, even for the goal of "expanding openSUSE's horizons".

> And I won't be asking openSUSE to morph into an Apache Foundation to enable any project either - I think it's a great community a great project for what it is.

Why not? openSUSE has considered itself, and  advertised itself as a community of projects for _years_

https://speakerdeck.com/sysrich/open-enterprise-and-open-community-working-together?slide=6 is a slide that I've been using since at least 2015 at SUSEcon

The Project has had sub-projects come and go with some frequency since that time.

If you or others in management have a preconceived notion that openSUSE is "just" a distribution Project, when the reality is the Project is already far more flexible than that, isn't it entirely possible that SUSE's plans for it's engagement (or lack thereof) with openSUSE could also be wildly off the mark?

Reply:
As you can see from the commit comment, it's considered a "clarification" by the author(s) and not a change of direction. The change is being done with the purpose of avoiding extensive interpretations. I always found the wording somewhat unfortunate, and the new one, although perhaps too detailed, fits SUSE's intent much better in my view.

And why I wouldn't be asking openSUSE to become the next Apache, LF, CNCF, TDF, Eclipse, FSF, Mozilla, ...? Simply because there are plenty of those already, and I do not see much value for the open source world in adding another one. Or for SUSE, for that matter.

Reply:
> As you can see from the commit comment, it's considered a "clarification" by the author(s) and not a change of direction. The change is being done with the purpose of avoiding extensive interpretations. I always found the wording somewhat unfortunate, and the new one, although perhaps too detailed, fits SUSE's intent much better in my view.

As you can see from the comments here, there are other stakeholders of the policy (external contributors, and internal contributors who commit signification amounts of their spare time in good faith) who clearly see this change as something different.

Regardless of intent, isn't how messages are understood that is important?

Even if I accept the intent, I still clearly see this as a change of direction - after all I can point to lots of supportive direction from management that supports other interpretations.

Unfortunate or not, we have to react to what we have actually said, what we have actually done, and deal with the consequences of how those words and actions have been perceived by the people impacted by them..surely?

> And why I wouldn't be asking openSUSE to become the next Apache, LF, CNCF, TDF, Eclipse, FSF, Mozilla, ...? Simply because there are plenty of those already, and I do not see much value for the open source world in adding another one. Or for SUSE, for that matter.

And yet, if the openSUSE community was naturally gravitating in that direction anyway, your decision to not support it is in itself a decision contrary to the communities wishes.

And that might be fine, that might not, again, it's an action that will be interpreted. My recurring theme in all my replies here is that we really should be comfortable "explaining our working" behind such decisions, wordings and actions, and not just "throw them over the wall" to our community and leave them to deal with the fallout.

Reply:
> But they're not the new horizons we talked about when discussing that all SUSE projects should go to Factory. That would be entirely new projects, not related to an operating system, coming under the openSUSE umbrella. So, @sysrich No, 14000 packages are not that, that's growth of the very same concept of packaging things into a distro. Kubic is perhaps the closest, as, while an OS again, it's one with a well defined usecase that's different of that of what Tumbleweed/Leap offer.

@vojtech-pavlik While I agree that we don't necessarily want to displace the FSF, ASF, etc., I think that projects that don't necessarily have a "predestined" upstream location should be able to find a home in the openSUSE Project. That enhances the brand and creates an opportunity to develop a halo effect for other projects under the openSUSE banner.

Reply:
> As you can see from the commit comment, it's considered a "clarification" by the author(s) and not a change of direction. The change is being done with the purpose of avoiding extensive interpretations. I always found the wording somewhat unfortunate, and the new one, although perhaps too detailed, fits SUSE's intent much better in my view.
> 
> And why I wouldn't be asking openSUSE to become the next Apache, LF, CNCF, TDF, Eclipse, FSF, Mozilla, ...? Simply because there are plenty of those already, and I do not see much value for the open source world in adding another one. Or for SUSE, for that matter.

The reality is, openSUSE is already much larger then just a couple of Linux distributions, as @sysrich mentioned there are already a number of other projects under the openSUSE Banner a more accurate description at the current time is "openSUSE is a collection of tools for sysadmins, developers and others" But I don't think it needs to be limited to that should other projects have good reasons to join. Generally the good reason in the past has been they intergrate closely with the creation or use of openSUSE Distro's think Yast, openQA and OBS, but at the same time several of the projects have become a part of openSUSE because they were started by existing openSUSE contributors and or SUSE employees during hackweek. 

SUSE has the power to help determine what openSUSE is through its contributions and if SUSE is going to contribute more thats great, but SUSE Management needs to understand that while the Linux Distributions may be the part of openSUSE that they are interested in growing it is not the only part of the community and other parts of the community are actively working to grow those area and projects. In the end the only people who can tell someone not to do something in openSUSE is the board and generally they can only do that if some form of conflict arises. If another project did want to join under the openSUSE banner as long as they are willing to follow openSUSE's Guidelines including being accountable to the board, its Codes of Conduct etc it then becomes a decision for the openSUSE Board, at this point the chairman could veto it but that would likely make them very unpopular with the rest of the community. 

I also don't think openSUSE needs to become the next FSF, ASF etc. but there may continue to be other projects that have certain synergies with openSUSE to the point where it makes sense then I think we should welcome them with open arms. When we have talked about an "openSUSE Foundation" in the past we are not talking about becoming another FSF or ASF etc we are talking about Setting up a Legal Structure to help grow mostly the Linux Distro part of openSUSE, as a Board we continue to hit things that limit our ability to grow there areas due to a lack of being able to do basic things such as accept donations both of Money and physical equipment this has impacted our ability to partner with things such as GSoC that could undoubtably help grow the distro side of openSUSE, but I think I need to send you a more detailed email with regards to this aspect as its somewhat offtopic from the original discussion.

But to summarise SUSE Management doesn't need to contribute to the non distro parts of openSUSE (Despite the fact they are major contributors to many of them) but it does need to accept that they two are parts of the openSUSE community that we as the openSUSE community value immensely

Reply:
> It is not only recommended for things tightly connected to SUSE Linux Enterprise, but in my opinion it should also clearly state that it is mandatory for submissions to the SLE code base. Without following Factory First on SLE submissions the submission will be rejected, with the exception of very few submissions (like for example Metadata packages)

I don't think we need to state that in the outward-facing Open Source Policy. What you describe is a policy on SLE development, not positioning of SUSE within the open source world.